src/ucl_util.c

```diff
@@ -3594,9 +3594,10 @@ ucl_object_copy_internal (const ucl_object_t *other, bool allow_array)
 
 		/* deep copy of values stored */
 		if (other->trash_stack[UCL_TRASH_KEY] != NULL) {
-			new->trash_stack[UCL_TRASH_KEY] =
-					strdup (other->trash_stack[UCL_TRASH_KEY]);
+			new->trash_stack[UCL_TRASH_KEY] = NULL;
 			if (other->key == (const char *)other->trash_stack[UCL_TRASH_KEY]) {
+				new->trash_stack[UCL_TRASH_KEY] = malloc(other->keylen);
+				memcpy(new->trash_stack[UCL_TRASH_KEY], other->trash_stack[UCL_TRASH_KEY], other->keylen);
```

Comment:
This change is really good, however, I think it will still be a good idea to terminate `key` with `\0`. Otherwise, it will definitely break functions like `ucl_object_key` which just return a `const char*`, e.g. how it is done in `ucl_copy_value_trash` function.

Reply:
This makes a lot of sense. I added the null-byte terminator in 110428d.